src/player.h

```diff
@@ -1503,6 +1496,28 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         bool has_fire(const int quantity) const;
         void use_fire(const int quantity);
 
+        /** Determine player's capability of recharging their CBMs. */
+        bool can_feed_battery_with( const item &it ) const;
+        bool can_feed_reactor_with( const item &it ) const;
+        bool can_feed_furnace_with( const item &it ) const;
+        /**
+         * Recharge CBMs whenever possible.
+         * @return true when recharging was successful.
+         */
+        bool feed_battery_with( item &it );
+        bool feed_reactor_with( item &it );
+        bool feed_furnace_with( item &it );
+        /** Check whether player can consume this very item */
+        bool can_consume_as_is( const item &it ) const;
+        /** Returns a reference to the argument. Use with care! */
+        item &as_comestible( item &it ) const;
```

Comment:
The function name and comment are very unhelpful.
From the code, it looks like it returns the item (if it's comestible), contents (if they're comestibles) or null item.
Would be more readable as `get_comestible` or `find_comestible`.